licensing/src/main/java/io/crate/license/LicenseService.java

```diff
@@ -168,7 +190,21 @@ public void clusterChanged(ClusterChangedEvent event) {
 
         if (newLicenseKey != null && !newLicenseKey.equals(previousLicenseKey)) {
             try {
-                this.currentLicense.set(licenseData(decodeLicense(newLicenseKey)));
+                DecryptedLicenseData decryptedLicenseData = licenseData(decodeLicense(newLicenseKey));
+                LicenseExpiryNotification expiryNotification = getLicenseExpiryNotification(decryptedLicenseData);
+
+                if (expiryNotification != null) {
+                    long millisToExpiration = decryptedLicenseData.millisToExpiration();
+                    if (expiryNotification.equals(LicenseExpiryNotification.EXPIRED)) {
+                        logger.error(expiryNotification.notificationMessage(millisToExpiration));
+                    } else if (expiryNotification.equals(LicenseExpiryNotification.SEVERE)) {
+                        logger.error(expiryNotification.notificationMessage(millisToExpiration));
+                    } else if (expiryNotification.equals(LicenseExpiryNotification.MODERATE)) {
+                        logger.warn(expiryNotification.notificationMessage(millisToExpiration));
+                    }
+                }
+
+                this.currentLicense.set(decryptedLicenseData);
             } catch (IOException e) {
                 logger.error("Received invalid license. Unable to read the license data.");
                 throw new InvalidLicenseException("Invalid license present in cluster");
```

Comment:
It's unrelated to this PR - but maybe for a follow up - throwing the exception here will also result in duplicate log entries.

Reply:
ah yes, will drop it here as need to squash anyway, good catch